lib/charms/hydra/v0/hydra_endpoints.py

```diff
@@ -94,7 +94,7 @@ def send_endpoint_relation_data(self, admin_endpoint: str, public_endpoint: str)
         if not self._charm.unit.is_leader():
             return
 
-        relations = self.model.relations[RELATION_NAME]
+        relations = self.model.relations[self._relation_name]
```

Comment:
```suggestion
        relations = self.model.relations.get(self._relation_name)
        if not relations:
            return
```

Let's also check that the relation exists.